File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.go
func productExceptSelf(nums []int) []int {
    res := make([]int, len(nums), len(nums))

    var reg int = 1
    //prefix
    for i, n := range nums {
        res[i] = reg
        reg = reg*n
    }
    //reset the register
    reg = 1
    //postfix
    for i := len(nums)-1; i >= 0; i-- {
        res[i] *= reg
        reg = reg*nums[i]
    }
    return res
}

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0]
        offset = 1
        for i in range(1, n + 1):
            if offset * 2 == i:
                offset = i
            res.append(1 + res[i - offset])
        return res
